inspect_byproducts.py

```diff
@@ -0,0 +1,163 @@
+"""
+<Program Name>
+  inspect_byproducts.py
+
+<Author>
+  Sachit Malik <i.sachitmalik@gmail.com>
+
+<Started>
+  June 6, 2017
+
+<Copyright>
+  See LICENSE for licensing information.
+
+<Requires>
+  in-toto
+
+<Purpose>
+  Inspections constitute an important part of in-toto. The script take the path
+  of a link file, a random string and an operator, which is used to compare
+  a certain property of the link file with the random string.
+
+  Suppose the link file is located at /user/abc/def/package.45gh325.link
+  and the user wants to check whether for the corresponding step (hence the link file),
+  the corresponding stderr field contains the string "test".
+
+  The usage would be as follows:
+  python inspect_byproducts.py -l  /user/abc/def/package.45gh325.link  -t stderr -o contains test
+
+  General usage:
+  python inspect_byproducts.py -l  <path/to/link/metadata/file>  -t <stdout|stderr>
+    -o [ is | is not | contains | contains not] <string to be tested>
+"""
+
+import os
+import sys
+import argparse
+import in_toto.log
+from in_toto.models.link import Link as link_import
+import securesystemslib.exceptions
+
+
+def inspect_byproducts(link, type, operator, input_string):
+    """
+    <Purpose>
+    A function which performs the inspection as described above depending on
+    various arguments.
+
+    <Arguments>
+      link:
+        the path to the link file
+
+      std:
+        whether to check stdout or stderr field
+
+      operator:
+        is | is not | contains | contains not
+
+      inputstring:
+        the string to be checked
+
+    <Exceptions>
+      Raises KeyError, in case the field corresponding to the key
+      in the dictionary is empty
+
+    <Returns>
+      Boolean
+    """
+
+    imported_link = link_import.read_from_file(link)
+
+    std_out_err = imported_link.byproducts[type]
+
+    if operator == 'is':
+      if std_out_err == input_string:
+        return True
+
+    elif operator == 'is not':
+      if std_out_err != input_string:
+        return True
+
+    elif operator == 'contains':
+      if std_out_err.find(input_string) != -1:
+        return True
+
+    elif operator == 'contains not':
+      if std_out_err.find(input_string) == -1:
+        return True
+
+    return False
+
+
+def parse_args():
+    """
+    <Purpose>
+      A function which parses the user supplied arguments.
+
+    <Arguments>
+      None
+
+    <Exceptions>
+      None
+
+    <Returns>
+      Parsed arguments (args object)
+    """
+    parser = argparse.ArgumentParser(
+        description="Inspects the byproducts of a step")
+
+    lpad = (len(parser.prog) + 1) * " "
+
+    parser.usage = ("\n"
+                    "%(prog)s --link <path to link metadata>\n{0}"
+                    "[--type of field to be inspected (--stdout | --stderr)]\n{0}"
+                    "[--operator (--is | --is not | --contains | --"
```

Comment:
I think it is better to not have whitespaces in the operator "words", because they'd have to be escaped. Maybe you can change to `is-not` and `contains-not`.

Also change the usage message to:
`--operator (is | is-not | contains | contains-not)`

Alternatively, as mentioned before you could make a [mutually exclusive option group](https://docs.python.org/2.7/howto/argparse.html#conflicting-options) for the operators.